src/main/java/seedu/address/logic/parser/RelateCommandParser.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.model.person.Id.isValidId;
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.RelateCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.IdContainsDigitsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindNumCommand object
+ */
+public class RelateCommandParser implements Parser<RelateCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RelateCommand
+     * and returns a RelateCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RelateCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RelateCommand.MESSAGE_USAGE));
+        }
+
+        String[] providedIDs = trimmedArgs.split("\\s+");
+
+        // Check if all IDs in input are valid
+        for (String iDInList : providedIDs) {
+            if (!isValidId(Integer.parseInt(iDInList))) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, RelateCommand.MESSAGE_USAGE));
+            }
+        }
+
+        /*
+         A decision is made here to convert the String[] into an Integer[] since an array of IDs should not contain
+         Strings, and this assumption is heavily relied upon in the following IDContainsDigitsPredicate class.
+         This is a potential bug as the provided IDs is not checked for existence here before being converted to an
+         Integer array. It also should not present as an efficiency issue due to small size of the array.
+        */
+        Integer[] providedIDsAsInt = Arrays.stream(providedIDs).map(Integer::parseInt).toArray(Integer[]::new);
+
+        return new RelateCommand(new IdContainsDigitsPredicate(Arrays.asList(providedIDsAsInt)));
```

Comment:
Understand that the future `showrelated` command will use IdContainsDigitsPredicate, but can I ask if there is a need for `RelateCommand` to take in `IdContainsDigitsPredicate`? Seems like only the two ids (as possibly integers) are needed.

Reply:
Do correct me if I'm wrong, but to my knowledge the only way to amend the viewed filter list is through the use of a predicate. The `ModelManager#getFilteredPersonList()` uses a predicate as well, and its implementation calls the `#setPredicate()` method on the `FilteredList<Person>` in `ModelManager`.